File: mybatis-flex-spring/src/main/java/com/mybatisflex/test/entity/Account.java
package com.mybatisflex.test.entity;

import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.KeyType;
import com.mybatisflex.annotation.Table;
import com.mybatisflex.core.keygen.KeyGenerators;
import lombok.Data;

import java.util.Date;

@Data
@Table("tb_account")
public class Account {

    @Id(keyType = KeyType.Generator, value = KeyGenerators.snowFlakeId)
    private Long id;
    private String userName;
    private Integer age;
    private Date birthday;
}

File: mybatis-flex-spring-kotlin/src/main/kotlin/com/mybatisflux/entity/Account.kt
package com.mybatisflux.entity

import com.mybatisflex.annotation.*
import com.mybatisflex.core.BaseMapper
import com.mybatisflex.core.activerecord.MapperModel
import com.mybatisflex.core.keygen.KeyGenerators
import com.mybatisflex.kotlin.extensions.db.mapper
import java.util.*

@Table("tb_account")
data class Account(
    @Id(keyType = KeyType.Generator, value = KeyGenerators.snowFlakeId)
    var id: Long? = null,
    var userName: String? = null,
    var age: Int? = null,
    var birthday: Date? = null,
    @Column(ignore = true)
    var maxAge: Int? = null,

    @RelationOneToOne(selfField = "id", targetField = "accountId")
    var idCard: IDCard? = null,
) : MapperModel<Account> {
    companion object : BaseMapper<Account> by mapper()


}

fun main() {
    Account.selectAll()
}
